File: src/leetcode/python/LinkedListCycle_141.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None


class LinkedListCycle_141(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        if head is None:
            return False
        slow = head
        while head is not None and head.next is not None:
            head = head.next.next
            slow = slow.next
            if head == slow:
                return True
        return False

File: src/leetcode/java/LinkedListCycle_141.java
package leetcode.java;

public class LinkedListCycle_141 {
  public boolean hasCycle(ListNode head) {
    if (head == null) return false;
    ListNode slow = head;
    while (head != null && head.next != null) {
      head = head.next.next;
      slow = slow.next;
      if (head == slow) return true;
    }
    return false;
  }

  class ListNode {
    int val;
    ListNode next;
    ListNode(int x) {
      val = x;
      next = null;
    }
  }
}
